test/nestedJoin.integration.test.ts

```diff
@@ -0,0 +1,432 @@
+            nestedJoin,
+            nestedPoolFromApi,
+            client,
+        });
+    });
+
+    test('native asset join', async () => {
+        const amountsIn = [
+            {
+                address:
+                    '0xc02aaa39b223fe8d0a0e5c4f27ead9083c756cc2' as Address, // WETH
+                rawAmount: parseUnits('1', 18),
+                decimals: 18,
+            },
+            {
+                address:
+                    '0x6b175474e89094c44da98b954eedeac495271d0f' as Address, // DAI
+                rawAmount: parseUnits('1', 18),
+                decimals: 18,
+            },
+            {
+                address:
+                    '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48' as Address, // USDC
+                rawAmount: parseUnits('1', 6),
+                decimals: 6,
+            },
+            {
+                address:
+                    '0xdac17f958d2ee523a2206206994597c13d831ec7' as Address, // USDT
+                rawAmount: parseUnits('1', 6),
+                decimals: 6,
+            },
+        ];
+
+        await doTransaction({
+            amountsIn,
+            chainId,
+            rpcUrl,
+            testAddress,
+            nestedJoin,
+            nestedPoolFromApi,
+            client,
+            useNativeAssetAsWrappedAmountIn: true,
+        });
+    });
+});
+
+export const doTransaction = async ({
+    amountsIn,
+    chainId,
+    rpcUrl,
+    testAddress,
+    nestedJoin,
+    nestedPoolFromApi,
+    client,
+    useNativeAssetAsWrappedAmountIn = false,
+}: TxInput) => {
+    const joinInput: NestedJoinInput = {
+        amountsIn,
+        chainId,
+        rpcUrl,
+        accountAddress: testAddress,
+        useNativeAssetAsWrappedAmountIn,
+    };
+    const queryResult = await nestedJoin.query(joinInput, nestedPoolFromApi);
+
+    // build join call with expected minBpOut based on slippage
+    const slippage = Slippage.fromPercentage('1'); // 1%
+
+    const signature = await Relayer.signRelayerApproval(
+        BALANCER_RELAYER[chainId],
+        testAddress,
+        client,
+    );
+
+    const { call, to, value, minBptOut } = nestedJoin.buildCall({
+        ...queryResult,
+        chainId,
+        slippage,
+        sender: testAddress,
+        recipient: testAddress,
+        relayerApprovalSignature: signature,
+    });
+
+    let tokensIn = joinInput.amountsIn.map(
+        (a) => new Token(chainId, a.address, a.decimals),
+    );
+    if (useNativeAssetAsWrappedAmountIn) {
+        tokensIn = replaceWrapped(tokensIn, chainId);
+    }
+
+    // send join transaction and check balance changes
+    const { transactionReceipt, balanceDeltas } =
+        await sendTransactionGetBalances(
+            [
+                ...tokensIn.map((t) => t.address),
+                queryResult.bptOut.token.address,
+            ],
+            client,
+            testAddress,
+            to,
+            call,
+            value,
+        );
+
```

Comment:
Should add a test for `value`. Mainly to make sure there is no value set when it shouldn't be as this would be bad.